src/app/pages/utils/form-utils.ts

```diff
@@ -0,0 +1,110 @@
+    console.log(errors)
+    for (const key of Object.keys(errors)) {
+      switch (key) {
+        case 'required':
+          return 'Este campo es requerido';
+
+        case 'minlength':
+          return `Debe ingresar al menos ${errors['minlength'].requiredLength} caracteres`;
+
+        case 'maxlength':
+          return `Debe ingresar como máximo ${errors['maxlength'].requiredLength} caracteres`;
+
+        case 'min':
+          return `Este campo debe ser mayor o igual ${errors['min'].min}`;
+
+        case 'email':
+          return 'El correo no es válido';
+
+        case 'pattern':
+            if (errors['pattern'].requiredPattern === FormUtils.firstNamePattern || errors['pattern'].requiredPattern === FormUtils.lastNamePattern) {
+              return `El ${inputName} no es válido`;
+            }
+
+          if(errors['pattern'].requiredPattern === FormUtils.emailPattern) {
+            return 'El correo electronico no es válido';
+          }
+
+          if(errors['pattern'].requiredPattern === FormUtils.notSpacesPattern) {
+            return 'Este campo no puede contener espacios';
+          }
+          if (errors['pattern'].requiredPattern === FormUtils.phoneNumberPattern) {
+            return 'El número de teléfono no es válido';
+          }
+          if (errors['pattern'].requiredPattern === FormUtils.cuitPattern) {
+            return 'El CUIT no es válido';
+          }
+          if (errors['pattern'].requiredPattern === FormUtils.dniPattern) {
+            return 'El DNI no es válido';
+          }
+          if (errors['pattern'].requiredPattern === FormUtils.datePattern) {
+            return 'La fecha de nacimiento no es válida';
+          }
+          
+          return 'Error de patrón contra expresión regular'
+
+        case 'emailTaken':
+          return 'El correo electrónico ya está siendo utilizado'
+
+        case 'passwordMismatch':
+          return 'Las contraseñas no coinciden';
+
+        default:
+          return `Error de validación no controlado ${key}`;
+
+      }
+    }
+    return null;
+  }
+
+  static passwordMatchValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
+    const password = control.get('password');
+    const confirmPassword = control.get('confirmPassword');
+    if (!password || !confirmPassword) {
+      return null;
+    }
+
+    const passwordsMatch = password.value === confirmPassword.value;
+
+
+    if (!passwordsMatch) {
+      confirmPassword.setErrors({ ...confirmPassword.errors, passwordMismatch: true });
+      return { passwordMismatch: true };
+    } else {
+
+      if (confirmPassword.errors?.['passwordMismatch']) {
+        const { passwordMismatch, ...otherErrors } = confirmPassword.errors;
+        const hasOtherErrors = Object.keys(otherErrors).length > 0;
+        confirmPassword.setErrors(hasOtherErrors ? otherErrors : null);
+      }
+    return null;
+    }
+  }
+}
```

Comment:
A esto lo veo un poco raro, creo que se podría simplificar

```suggestion
  static passwordMatchValidator: ValidatorFn = (control: AbstractControl): ValidationErrors | null => {
    const password = control.get('password');
    const confirmPassword = control.get('confirmPassword');
    if (!password || !confirmPassword) return null;
    return password.value === confirmPassword.value
        ? null
        : { passwordMismatch: true };
  }
```

Igual es una opinión, capaz se maneja otra lógica que desconozco.

Reply:
Esto que vos planteás no se puede hacer, porque el error de passwordMismatch es a nivel de form, y AngularMaterial solamente te muestra los errores que están asociados a un control, debiendo setearle el error del passwordMismatch a alguno de los control (en este caso el ConfirmPassword)